File: 455. Assign Cookies/C++/Solution.cpp
class Solution {
public:
    int findContentChildren(vector<int> &g, vector<int> &s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int i = 0;
        int count = 0;
        for (auto item : g) {
            while (i < s.size() && item > s[i]) {
                i++;
            }
            if (i == s.size()) {
                break;
            }
            count++;
            i++;
        }
        return count;
    }
};

File: 455. Assign Cookies/Java/Solution.java
public class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);
        int i = 0;
        int count = 0;
        for (int item : g) {
            while (i < s.length && item > s[i]) {
                i++;
            }
            if (i == s.length) {
                break;
            }
            count++;
            i++;
        }
        return count;
    }
}